src/ekat/util/ekat_meta_utils.hpp

```diff
@@ -7,34 +7,25 @@
 namespace ekat
 {
 
-// A parameter pack. To be used for template specializations
+// A list of types is just an alias for tuple
 template<typename... Ts>
-struct TypeList;
+using TypeList = std::tuple<Ts...>;
 
-template<typename T>
-struct TypeList<T> {
-  using head = T;
-  static constexpr int size = 1;
-};
-
-template<typename T, typename... Ts>
-struct TypeList<T,Ts...> {
-private:
-  using tail = TypeList<Ts...>;
-public:
-  using head = T;
-  static constexpr int size = 1 + tail::size;
-};
+template<typename TL>
+struct type_list_size : std::tuple_size<TL> {};
 
-// Cat two packs
-template<typename P1, typename P2>
-struct CatLists;
+// Cat two type lists
```

Comment:
Maybe the comment here could say "Concatenate" instead of "Cat", just to be clear.